From michael@jupiter.limit.org  Mon Aug 13 11:08:17 2001
X-VM-v5-Data: ([nil nil nil nil nil nil nil nil nil]
	["3203" "Monday" "13" "August" "2001" "11:03:16" "-0700" "Michael Matsumura" "michael@limit.org" nil "100" "er. the file." "^From:" nil nil "8" nil nil nil nil nil nil nil nil]
	nil)
Return-Path: <michael@jupiter.limit.org>
Received: from localhost (IDENT:root@goliath.my-net [192.168.1.2])
	by cx81598-b.chnd1.az.home.com (8.11.2/8.11.2) with ESMTP id f7DI8H527354
	for <edwinh+flexbackup@mailhost>; Mon, 13 Aug 2001 11:08:17 -0700
Received: from mail.chnd1.az.home.com
	by localhost with POP3 (fetchmail-5.7.4)
	for edwinh+flexbackup@mailhost (single-drop); Mon, 13 Aug 2001 11:08:17 -0700 (MST)
Received: from mh12-sfba.mail.home.com ([24.0.95.247])
          by femail30.sdc1.sfba.home.com
          (InterMail vM.4.01.03.20 201-229-121-120-20010223) with ESMTP
          id <20010813180345.NPUJ20694.femail30.sdc1.sfba.home.com@mh12-sfba.mail.home.com>
          for <flexbackup@mail.chnd1.az.home.com>;
          Mon, 13 Aug 2001 11:03:45 -0700
Received: from mx12-sfba.mail.home.com (mx12-sfba.mail.home.com [24.0.95.243])
	by mh12-sfba.mail.home.com (8.9.3/8.9.0) with ESMTP id LAA02534
	for <flexbackup@home.com>; Mon, 13 Aug 2001 11:03:45 -0700 (PDT)
Received: from jupiter.limit.org (limit.org [216.102.231.2])
	by mx12-sfba.mail.home.com (8.11.1/8.11.1) with ESMTP id f7DI3eM12679
	for <flexbackup@home.com>; Mon, 13 Aug 2001 11:03:44 -0700 (PDT)
Received: (from michael@localhost)
	by jupiter.limit.org (8.11.4/8.11.4) id f7DI3Ho07005
	for flexbackup@home.com; Mon, 13 Aug 2001 11:03:17 -0700 (PDT)
	(envelope-from michael)
Message-ID: <20010813110316.A6997@jupiter.limit.org>
Mime-Version: 1.0
Content-Type: multipart/mixed; boundary="VbJkn9YxBvnuCH5J"
Content-Disposition: inline
Status: O
Content-Length: 3203
Lines: 100
From: Michael Matsumura <michael@limit.org>
To: flexbackup@home.com
Subject: er. the file.
Date: Mon, 13 Aug 2001 11:03:16 -0700
User-Agent: Mutt/1.2.5i


--VbJkn9YxBvnuCH5J
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

oops. forgot to attach the file. here it is

-- 
Michael Matsumura
michael@limit.org

--VbJkn9YxBvnuCH5J
Content-Type: text/plain; charset=us-ascii
Content-Disposition: attachment; filename="flexbackup.patch"

--- /usr/bin/flexbackup.old	Mon Aug 13 09:17:56 2001
+++ /usr/bin/flexbackup	Mon Aug 13 10:32:47 2001
@@ -144,7 +144,11 @@
 &set_tape_operation_defaults();
 
 # Get current date string
-$main::date = &current_time("short");
+if ($cfg::staticfiles eq "false") {
+    $main::date = &current_time("short");
+} else {
+    $main::date = "";
+}
 
 # Decide what to do
 if (defined($main::opt{'restore'})) {
@@ -208,7 +212,14 @@
 
     # Figure out log file name & empty log file
     $label = &get_label($main::opt{'fs'});
-    $logfile = "$cfg::prefix$label.$main::level.$main::date";
+    if ($cfg::staticfiles eq "false") {
+        $logfile = "$cfg::prefix$label.$main::level.$main::date";
+    } else {
+        $logfile = "$cfg::prefix$label.$main::level";
+	if (-f $logfile) {
+            rename($logfile, $logfile.".old");
+	}
+    }
     $symlink = "$cfg::prefix$label.latest";
     $main::log = "$cfg::logdir/$logfile";
     if (! open(LOG,">$main::log")) {
@@ -392,6 +403,9 @@
             $logext = ".Z";
             $comp_cmd = "compress -f $main::log";
         }
+	if (-f $logfile . $logext) {
+            rename($logfile . $logext, $logfile . $logext . ".old");
+	}
         &line("logonly");
         print "| Compressing log ($logfile" . "$logext)\n";
         system("$comp_cmd");
@@ -446,7 +460,11 @@
     # Create file name if writing to a file
     # (config file's $device points to a dir in this case)
     if (defined($main::use_file)) {
-        $_ = $main::level . "." . $main::date . "." . $cfg::type;
+        if ($cfg::staticfiles eq "false") {
+            $_ = $main::level . "." . $main::date . "." . $cfg::type;
+        } else {
+            $_ = $main::level . "." . $cfg::type;
+        }
         if ($cfg::type =~ m/^(tar|dump|cpio)$/) {
             if ($cfg::compress eq "gzip") {
                 $_ .= ".gz";
@@ -467,6 +485,9 @@
             }
         }
         $main::device = $cfg::device . "/" . $label . "." . $_;
+        if ($cfg::staticfiles eq "true" && -f $main::device) {
+            rename($main::device, $main::device.".old");
+	}
     }
 
     # Just get the date for now; don't write the timestamp
@@ -1448,6 +1469,7 @@
     &check(\$cfg::verbose,'verbose');
     &check(\$cfg::sparse,'sparse');
     &check(\$cfg::indexes,'indexes');
+    &check(\$cfg::staticfiles,'staticfiles');
     &check(\$cfg::buffer,'buffer');
     &check(\$cfg::buffer_megs,'buffer_megs','exist');
     &check(\$cfg::pad_blocks,'pad_blocks');
@@ -1884,6 +1906,10 @@
     if ($cfg::indexes eq "true") {
         tie(%main::index,"AnyDBM_File",$cfg::index,O_CREAT|O_RDWR,0640) or
             push(@main::errors,"Can't tie DB $cfg::index: $OS_ERROR");
+    }
+
+    if ($cfg::staticfiles ne "true" && $cfg::staticfiles ne "false") {
+        push(@main::errors,"\$staticfiles must be 'true' or 'false': $cfg::staticfiles");
     }
 
     # Levels

--VbJkn9YxBvnuCH5J--

